src/org/omegat/filters2/master/PluginUtils.java

```diff
@@ -167,6 +180,9 @@ public static void loadPlugins(final Map<String, String> params) {
                 Log.log(ex);
             }
         }
+
+        checkForLatestPluginVersion(urlList);
```

Comment:
method `checkForLatestPluginVersion` manipulate the `urlList` .  I think it is better to have a method `populatePluginUrlList()` that returns `List<URL>`. It may be like

```
protected static List<URL> populatePluginUrlList() {
        // list all jars in /plugins/
        FileFilter jarFilter = pathname -> pathname.getName().endsWith(".jar");
        List<File> fs = pluginsDirs.stream().flatMap(dir -> FileUtil.findFiles(dir, jarFilter).stream())
                .collect(Collectors.toList());
        List<URL> urlList = new ArrayList<>();
        for (File f : fs) {
            try {
                URL url = f.toURI().toURL();
                urlList.add(url);
                Log.logInfoRB("PLUGIN_LOAD_JAR", url.toString());
            } catch (IOException ex) {
                Log.log(ex);
           }
        }

       // body of checkForLatestPluginVersion here..
  
       return urlList;
```

Reply:
Good idea, that should be cleaner.

Done in 183ad9eef4ff0a6e576f269d6931f8d3e1fd610a